src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunction.cs

```diff
@@ -0,0 +1,262 @@
+            private readonly TimeZoneInfo _timeZoneInfo;
+
+            internal JsonProcessing(TimeZoneInfo timezoneInfo, IRContext irContext, FormulaValue[] args)
+            {
+                _args = args;
+                _irContext = irContext;
+                _timeZoneInfo = timezoneInfo;
+            }
+
+            internal FormulaValue Process()
+            {
+                GetFlags();
+
+                if (_flags == null || HasUnsupportedType(_args[0].Type._type, out _, out _))
+                {
+                    return CommonErrors.GenericInvalidArgument(_irContext);
+                }
+
+                using MemoryStream memoryStream = new MemoryStream();
+                using Utf8JsonWriter writer = new Utf8JsonWriter(memoryStream, new JsonWriterOptions() { Indented = _flags.IndentFour });
+
+                WriteToJson(writer, _args[0]);
+                writer.Flush();
+
+                string json = Encoding.UTF8.GetString(memoryStream.ToArray()).Replace(@"\u0022", @"\""").Replace(@"\u0027", @"'");
+
+                // replace two spaces with four spaces only at the beginning of each line
+                json = Regex.Replace(json, @"^(?<spc>(  )+)(?<right>[^ ].*)", @"$2$2$3", RegexOptions.Multiline);
+
+                return new StringValue(_irContext, json);
+            }
+
+            private void GetFlags()
+            {
+                _flags = new JsonFlags() { HasMedia = DataHasMedia(_args[0].Type._type) };
+
+                if (_args.Length > 1 && _args[1] is StringValue arg1string)
+                {
+                    _flags.IgnoreBinaryData = arg1string.Value.Contains("G");
+                    _flags.IgnoreUnsupportedTypes = arg1string.Value.Contains("I");
+                    _flags.IncludeBinaryData = arg1string.Value.Contains("B");
+                    _flags.IndentFour = arg1string.Value.Contains("4");
+                }
+
+                if (_args.Length > 1 && _args[1] is OptionSetValue arg1optionset)
+                {
+                    _flags.IgnoreBinaryData = arg1optionset.Option == "IgnoreBinaryData";
+                    _flags.IgnoreUnsupportedTypes = arg1optionset.Option == "IgnoreUnsupportedTypes";
+                    _flags.IncludeBinaryData = arg1optionset.Option == "IncludeBinaryData";
+                    _flags.IndentFour = arg1optionset.Option == "IndentFour";
+                }
+
+                if ((_flags.IncludeBinaryData && _flags.IgnoreBinaryData) ||
+                    (_flags.HasMedia && !_flags.IncludeBinaryData && !_flags.IgnoreBinaryData) ||
+                    (!_flags.IgnoreUnsupportedTypes && HasUnsupportedType(_args[0].Type._type, out var _, out var _)))
+                {
+                    _flags = null;
+                    return;
+                }
+            }
+
+            internal void WriteToJson(Utf8JsonWriter writer, FormulaValue arg)
+            {
+                if (arg is BlankValue blankValue)
```

Comment:
>arg [](http://example.com/codeflow?start=20&length=3)

consider IValueVisitor here instead of switch.
That will ensure we catch all cases. 
(We can not-impl unsupported things)